src/main/java/ca/corefacility/bioinformatics/irida/model/sample/metadata/PipelineProvidedMetadataEntry.java

```diff
@@ -43,6 +48,21 @@ public PipelineProvidedMetadataEntry(String value, String type, AnalysisSubmissi
 	public AnalysisSubmission getSubmission() {
 		return submission;
 	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void merge(MetadataEntry metadataEntry) {
+		checkArgument(metadataEntry instanceof PipelineProvidedMetadataEntry,
+				"You cannot merge a non-PipelineProvidedMetadataEntry into a PipelineProvidedMetadataEntry: metadataEntry=" + metadataEntry);
```

Comment:
Can you move this check up to `MetadataEntry` and make it general for any subclass?  It should likely explode if any different class of MetadataEntry is being merged here.  Just for a 2nd level check to what you're doing within Sample.

Reply:
I could, but then I couldn't call `super.merge()` in `PipelineProvidedMetadataEntry` (as it would fail since it's not the same class).  I could just repeat the code in `MetadataEntry` here though if that's what you want?

Reply:
Maybe the logic would need to be changed a bit.  Instead of 
```java
metadataEntry instanceof PipelineProvidedMetadataEntry
``` 
maybe, more like
```java
metadataEntry.getClass().isAssignableFrom(this.getClass())
```
That way a `PipelineProvidedMetadataEntry` could merge into another `PipelineProvidedMetadataEntry`, but not `MetadataEntry` into `PipelineProvidedMetadataEntry`.  I think that's the right order, you might need to invert `metadataEntry` and `this` in the above code block.

Reply:
Thanks. I did not know about that method. Added in a1620b243b15b2f35c38ec1c19e89520bb652380

Reply:
I've fixed this in 6acff1306bc7a334fe3fbabbea691b8b605f6bcc.

However, I had to change the code to not use `isAssignableFrom` as it won't actually work (`MetadataEntry` is assignable from `PipelineProvidedMetadataEntry`).

Instead, I just directly check if classes are equal in `MetadataEntry` and don't call the superclass method in `PipelineProvidedMetadataEntry`.

Reply:
After thinking about this a bit more, I can actually use the `super.merge()` method **and** remove the class check in `PipelineProvidedMetadataEntry`, just not with `isAssignableFrom`. I have to use `Class.equals()`. Fixed in 1d63f2e9977d35341472d0993e0f5936ecabb4b5

Reply:
I was going to ask you about that this morning.  This should work.  Thanks.